File: objectiveai-rs/src/agent/completions/request/agent_completion_create_params.rs
//! Agent completion request parameters.

use indexmap::IndexMap;
use serde::{Deserialize, Serialize};
use schemars::JsonSchema;

/// Parameters for creating a agent completion.
#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
#[schemars(rename = "agent.completions.request.AgentCompletionCreateParams")]
pub struct AgentCompletionCreateParams {
    /// The conversation messages.
    pub messages: Vec<super::super::message::Message>,
    /// Provider routing preferences.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub provider: Option<super::Provider>,
    /// The agent to use (inline Agent or stored ID).
    pub agent: crate::agent::InlineAgentBaseWithFallbacksOrRemote,
    /// Output format constraints (text, JSON, or JSON schema).
    #[serde(skip_serializing_if = "Option::is_none")]
    pub response_format: Option<super::ResponseFormatParam>,
    /// Random seed for deterministic generation.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub seed: Option<i64>,
    /// Whether to stream the response.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub stream: Option<bool>,

    // --- MCP server authorization ---
    /// Map from MCP server URL to authorization header value.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub mcp_server_authorization: Option<IndexMap<String, String>>,

}

File: objectiveai-js/src/agent/completions/request/agentCompletionCreateParams.ts
import { z } from "zod";
import { AgentCompletionsMessageMessageSchema } from "../message/message";
import { AgentCompletionsRequestAgentSchema } from "./agent";
import { AgentCompletionsRequestProviderSchema } from "./provider";
import { AgentCompletionsRequestResponseFormatParamSchema } from "./responseFormatParam";

export const AgentCompletionsRequestAgentCompletionCreateParamsSchema = z.object({
  messages: z.array(AgentCompletionsMessageMessageSchema).describe("The conversation messages."),
  provider: AgentCompletionsRequestProviderSchema.nullable().describe("Provider routing preferences.").optional(),
  agent: AgentCompletionsRequestAgentSchema.describe("The agent to use (inline Agent or stored ID)."),
  agents: z.array(AgentCompletionsRequestAgentSchema).nullable().describe("Alternative agents to try if the primary agent fails.").optional(),
  response_format: AgentCompletionsRequestResponseFormatParamSchema.nullable().describe("Output format constraints (text, JSON, or JSON schema).").optional(),
  seed: z.number().int().meta({ format: "int64" }).nullable().describe("Random seed for deterministic generation.").optional(),
  stream: z.boolean().nullable().describe("Whether to stream the response.").optional(),
  mcp_server_authorization: z.record(z.string(), z.string()).nullable().describe("Map from MCP server URL to authorization header value.").optional(),
}).describe("Parameters for creating a agent completion.").meta({ title: "agent.completions.request.AgentCompletionCreateParams" });
export type AgentCompletionsRequestAgentCompletionCreateParams = z.infer<typeof AgentCompletionsRequestAgentCompletionCreateParamsSchema>;
